src/keydown_content.ts

```diff
@@ -1,119 +0,0 @@
-/** Shim for the keyboard API because it won't hit in FF57. */
-
-import * as Messaging from "./messaging"
-import * as msgsafe from "./msgsafe"
-import { isTextEditable, getAllDocumentFrames } from "./dom"
-import { isSimpleKey } from "./keyseq"
-
-function keyeventHandler(ke: KeyboardEvent) {
-    // Ignore JS-generated events for security reasons.
-    if (!ke.isTrusted) return
-
-    // Mode is changed based on ke target in the bg.
-    modeSpecificSuppression(ke)
-
-    Messaging.message("keydown_background", "recvEvent", [
-        msgsafe.KeyboardEvent(ke),
-    ])
-}
-
-// {{{ Bad key suppression system
-
-// This is all awful and will go away when we move the parsers and stuff to content properly.
-
-import state from "./state"
-
-import * as generic from "./parsers/genericmode"
-let normparser = keys => generic.parser("nmaps", keys)
-let inputparser = keys => generic.parser("inputmaps", keys)
-let insertparser = keys => generic.parser("imaps", keys)
-let ignoreparser = keys => generic.parser("ignoremaps", keys)
-let keys = []
-
-function genericResponse(ke: KeyboardEvent, parser: any): void {
-    keys.push(ke)
-    const response = parser(keys)
-
-    // Suppress if there's a match.
-    if (response.isMatch) {
```

Comment:
these lines were important - it's what made Tridactyl binds override Firefox binds, if they existed

Reply:
Key suppression regression looks to have been resolved. Moving on to fixing hint-mode so that I can dogfood it while hunting down the rest of the issues.